autonat/autonat-v2.md

```diff
@@ -0,0 +1,193 @@
+# AutonatV2: spec
```

Comment:
Should we handle addresses from private ranges like `10.x.y.z` in a dedicated way?

Two nodes might find themselves in a private network as reachable but they will still be behind NAT  in regards to the Internet.

https://en.m.wikipedia.org/wiki/Private_network

Reply:
I've added a suggestion for implementations to only verify private address reachability with nodes on the same subnet. 

Do you think the spec should have information regarding what to do with this information? 

Consider the case that the node is reachable on a 192.168.1.100 address within its subnet. Now this node should advertise this address to other nodes(via identify) in case eventually this address reaches a node on the same subnet(via DHT or some other mechanism). Is my understanding correct here?

Reply:
My thinking was that we discard any addresses that are by design not globally reachable / routeable. I am not sure it makes sense for nodes to use Auto**NAT** to determine reachability on the same subnet? What we care about is reachability from the Internet, is it not?

To discover nodes on the same subnet, nodes can for example use mDNS.

Reply:
done.